### PR TITLE
feat: make list public by default

### PR DESCRIPTION
**Pull request by kimlimjustin**
Adding features mentioned in #44. I've made the lists part to be public by default after initial creation, making non-authenticated users able to see it just by opening the same link as what the authenticated users use to view their lists.

In order make this possible, I've created a new struct called `Authenticated` which does not force-checking every request and send them to the login page if not authenticated, instead it just simply says that "this is not authenticated" if it does not authenticated while returning the `auth_user` properties if authenticated. 

Then, I added a new column to the `lists`, called `prvate` (not `private` or `priv` as they are keywords so yeah I chose some random key), which is set to false as default. Finally, it uses conditional rendering according to the key. I've also added a button so that users are able to toggle the publicity of the list.

### CODE REVIEW

File: src/routes/lists.rs [hunk 1]
```diff
@@ -65,11 +88,45 @@ async fn get_create(
...
+            if list.user_id != user.user_id {
+                return Err(ResponseError::NotFound)
+            }
+        },
+        None => {
+            return Err(ResponseError::NotFound)
+        }
+    }
+    
+    if list.prvate {
```
Comment (raffomania) [inline src/routes/lists.rs:123]: You can replace this whole block with:

```rs
db::lists::set_private(&mut tx, query.list_id, list.private).await?;
```

File: templates/list.html [hunk 2]
```diff
@@ -28,7 +29,21 @@ <h1 class="text-xl font-bold">{{ list.title }}</h1>
       hx-select="#create_link"
       >Link an item</a
     >
+    <a
```
Comment (raffomania) [inline templates/list.html:32]: After clicking this link, the whole section is empty. You'll need to add `id="toggle_publicity"`, remove the `hx-target` attribute and add `hx-swap="outerHTML"` to make it work.

**Review by raffomania — CHANGES_REQUESTED** [review]
Thanks for taking the time to implement this! I think there are some opportunities to streamline the code we need to get this feature.

A thing that would be really nice would be a checkbox in the `CreateList` form allowing users to choose between private and public lists when creating them. That's optional though, I'll merge this PR without that feature.

**Comment by kimlimjustin** [comment]
Hello @raffomania, I've made some updates for this PR, could you please take a look again? Thanks in advance

**Review by raffomania — APPROVED** [review]

**Pull request merged**